cranelift/filetests/filetests/isa/x64/struct-ret.clif

```diff
@@ -11,14 +11,12 @@ block0(v0: i64):
 ;   pushq   %rbp
 ;   movq    %rsp, %rbp
 ; block0:
-;   movq    %rdi, %rax
```

Comment:
Doesn't this now violate the x64 ABI?  I thought this requires the struct-ret pointer to be returned in %rax, that was the whole point of the logic of adding a `StructReturn` element to the return values?

Reply:
There are currently some problems with struct-return that I'm tracking down. I think all the comments you added are directly related.